model/plugin_kvset_options.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// PluginKVSetOptions contains information on how to store a value in the plugin KV store.
+type PluginKVSetOptions struct {
+	EncodeJSON      bool        // If true store the information encoded as JSON
+	Atomic          bool        // Only store the value if the current value matches the oldValue
+	OldValue        interface{} // The value to compare with the current value
```

Comment:
Alternatively, we don't need `Atomic`, and could automatically use compare-and-set semantics if `OldValue != nil`?

Reply:
But what if `OldValue` should be `nil` before we set the new value? E.g. make sure an information doesn't exist before you write it.